cmd/zapi/cmd/new/new.go

```diff
@@ -47,14 +51,28 @@ func (c *Command) Run(args []string) error {
 	if len(args) != 1 {
 		return errors.New("must specify a space name")
 	}
+
+	storageCfg := &storage.Config{
+		Kind: storage.Kind(c.kind),
+	}
+	if c.thresh != "" {
+		if thresh, err := units.ParseStrictBytes(c.thresh); err != nil {
+			return fmt.Errorf("invalid log-size-threshold: %w", err)
+		} else {
+			storageCfg.Archive = &storage.ArchiveConfig{
+				CreateOptions: &storage.ArchiveCreateOptions{
+					LogSizeThreshold: &thresh,
+				},
+			}
+		}
```

Comment:
If you keep this, I think it reads better if you don't wrap the success path inside an else block.
```suggestion
		thresh, err := units.ParseStrictBytes(c.thresh)
		if err != nil {
			return fmt.Errorf("invalid log-size-threshold: %w", err)
		}
		storageCfg.Archive = &storage.ArchiveConfig{
			CreateOptions: &storage.ArchiveCreateOptions{
				LogSizeThreshold: &thresh,
			},
		}
```